### DISCUSSION TITLE
Reset the EEPROM and cannot connect it with PC anymore

### DISCUSSION THREAD

**Discussion by buabula** [bitcraze/discussions#1302] [Q&A] [answered]
Hello everyone,
I recently encountered an issue with my Crazyflie Bolt after attempting to modify the radio channel settings using script. I changed the elems['radio_channel'] from 80 to 89 in the EEPROM settings via the write-eeprom.py script from the crazyflie-lib-python examples. Initially, I could still connect to the Crazyflie at radio://0/80/2M, but the "Configure 2.X" showed the channel as 89. It seemed like the channel change was only partially successful.

After turning off the device and turning it back on, I was no longer able to find the Crazyflie using the radio scan. Attempting to reset the EEPROM using a USB connection as suggested in the documentation ("reset the EEPROM"), I found that the script would not execute successfully.

Then I use the DFU to flash the firmware (2023.07),the client can scan radio://0/80/2M but cannot connect and I'm concerned I may have corrupted the EEPROM settings or incorrectly configured the device.

What steps can I take to restore the Crazyflie's and ensure it is discoverable again?
![1](https://github.com/bitcraze/discussions/assets/113815745/3ddf551b-8bf9-4b70-8c86-3f00f696b3d4)
![image](https://github.com/bitcraze/discussions/assets/113815745/259ec158-5c09-430f-b196-568635b06256)

Any help or suggestions would be greatly appreciated. Thank you for your assistance!

**Comment by buabula** [comment]
And the M2 is always light now, after I flash the firmware via DFU...

**Reply by ToveRumar** [reply]
Hi!

This sounds like something could be corrupted, just like you say. We don't advice to flash with DFU. 
However,  you could try reflashing via bootloader (recovery mode). Hopefully that should work, and then at least the bootloader is intact, and the CF is recoverable. 
Make sure to flash bolt FW. Loading the wrong FW in bootloader mode results in a freeze.

**Reply by buabula** [reply]
Thank you very much for your response! I am sorry for my mistakes caused by negligence. I have attempted to use the bootloader (recovery mode), the Bolt can connect with PC by pressing the "Initiate bootloader cold boot" . However, when I press the button "Program",  the Bolt simply restarts once and then shows as "not connected." I cannot continue to press the "restart in firmware mode". So I tried skipping the "Program" step and directly selected "restart in firmware mode.", the Bolt restart again , but it still fails to connect afterward.
I would greatly appreciate any guidance or suggestions you might have to help resolve this situation. Thank you for your patience and support!

**Reply by buabula** [reply]
And I have tried to flash another version firmware(this time 2023.11-bolt, last time 2024.02-bolt) via bootloader (recovery mode) , I see the process of flashing, but it still cannot establish the connect with PC, even cannot find radio number by pressing "scan".
In addition, I flash 2023.11-cf2/bolt via bootloader (recovery mode) ,it looks like as same as the beginning. ("scan" can find "radio://0/80/2M ", but cannot connect),the only constant light led is M2 when in normal mode.

**Reply by ToveRumar** [reply, marked as answer]
Hi!

Ok, great news then! It sounds like your bootloader is intact. 
It sounds like the setting in EEPROM settings is in some way corrupted. I would suggest trying to connect with your device over USB and change the address from there (using the cf client). Hopefully that helps!

If that does not work I think you need to reset the EEPROM and build it in custom FW and flash that. I can share a solution how to do that if it is not solved by the above method.

**Reply by buabula** [reply]
Thank you very much for your assistance! It works, and I can now successfully connect the Bolt to my PC. Although I'm not sure if it's appropriate to ask continually, I would like to understand how to reset the EEPROM. My initial issue stemmed from finding the instructions on the ["Reset"](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/development/eeprom/) page, and now I'm interested in learning how to properly perform this operation.
Thank you once again for your help!

**Reply by ToveRumar** [reply]
Hi!

Ok, that its great!
Im struggling a bit to find where it might have gone wrong, except for maybe flashing with DFU. 

If you reset the eeprom using the script, and then restart the device. Can you connect on the new channel then?

Reading the last log it looks more like your radio is disconnected in some way.

**Reply by buabula** [reply]
Hi！

Just after using the script, I can connect it, but the "Select an interface" text still read "radio://0/80/2M". I can also opened "_connect-configure 2.X", but the channel appeared as "89".  I think it is strange, so maybe I press the button "write" in configure 2.X. Once completed, I restarted the device but cannot scan for my device. 

The log shows ”cannot find a crazyradio“, which might be due to me forgetting to plug it in, but I'm not certain. In fact I haven't flashed anything in crazyradio. 

Additionally, could you tell me why the address is different from the default? I haven't changed the address before. I find flashing the firmware via bootloader will "[_flash everything but MBR and MBS_](https://www.bitcraze.io/documentation/repository/crazyflie2-nrf-firmware/master/development/architecture/)", but maybe the MBR and MBS aren't related with the address?

**Reply by ToveRumar** [reply]
Hi!

The address is not reset to default on flashing new FW. It is stored in EEPROM and this will remain the same throughout FW updates.

That the text still reads "radio://0/80/2M" sounds like you need to do a re-scan. Or maybe you have already done that? 

Have you retried doing all the steps again, without using DFU? 
If so, and the issue remains, could you please write all the steps down? I have not been able to recreate this, so there might be steps missing.

**Reply by gemenerik** [reply]
If you cannot find it anymore; connect your Crazyflie to your computer with a USB cable. Connect to it in cfclient. Then in 'connect'->'configure 2.x' you can set it to a known address. Now you should be able to find it with your Crazyradio.

> Additionally, could you tell me why the address is different from the default?

You may have (accidentally) changed the value in the 'configure 2.x' window. When you click 'write' this sets the address on the Crazyflie.

**Reply by buabula** [reply]
Thank you all very much! I'm sorry for my late reply. 

**For Tove,** actually I had tried to use DFU once, so maybe it is a possible reason why the address has been changed? 

For more descriptions:

Firstly, I used the "write-eeprom.py" script to change elems['radio_channel'] from 80 to 89. Before I restarted the device, the device can still connect with PC showing "89" in "Configure 2.X-channel" as well as "radio://0/80/2M" in "Select an interface" 。

Secondly, I restarted the bolt, then the radio was unable to scan the device (no radio number in "Select an interface"), so I tried to use DFU to flash FW (2023.07), the information shown I flashed it successfully. Although I can scan the device, showing radio://0/80/2M in "Select an interface", I cannot connect it successfully.

Thirdly, I followed your instructions using the bootloader (recovery mode) to flash FW (2023.11). Initially, it is still unable to connect (But I forgot the details...). Then I used the USB cable to find address in "Configure 2.X", it is not the default address but a new one (I have never seen it before). I changed the "0xE7E7E7E7E7" into new address (under the "Select an interface" and beyond the" Flight Control"), then I can find and connect with it.

**For gemenerik,** in fact, I have never changed the address before (in my mind), I don't know what address is suitable to set since I haven't researched swarms recently. Of course, maybe I have done it? But I forgot that totally. T - T

Thanks for your help again!

**Reply by buabula** [reply]
Or maybe I should start a new one question?

**Reply by ToveRumar** [reply]
Hi! 


It looks like you might have changed the address as well as the channel with the write_eeprom.py script. By default the script will program the address as 0xDEADBEEF. Then you will not be able to see it if you scan for 0xE7E7E7E7E7. Could that be the reason?
In that case you need to change the address in the address field to be able to find it when scanning. 

I have not been able to reproduce this. So it would be helpful if you could re-try the procedure with very clear steps and come back if it does not work this time around either.

**Reply by buabula** [reply]
Yes! The new address is 0xDEADBEEF, I think it is the reason as you say! I thought the default address is 0xE7E7E7E7E7，and I haven't specifically modified the address, so I missed the script. Thank you very much for your help! 
﻿
But why the address in script is not the 0xE7E7E7E7E7? All bolts I got have the same initial address 0xE7E7E7E7E7.

**Reply by knmcguire** [reply]
You are right, probably this should be changed to another address, however this script was not meant as a recovery script but just as an example of how to change the eeprom.

Since you've already selected an answer to discussion, if you have anything else that you need help with, could you start a new discussion thread? This will give us a better oversight

**Reply by buabula** [reply]
Of course! Thanks for your help and patience!
